tests/st/distributed/test_l2_multi_orch.py

```
@@ -0,0 +1,226 @@
+
+        # 3) Per-orch metadata is correct (both orchs share the same signature here)
+        assert add_via_sub.param_names == ["a", "b", "f"]
+        assert sub_via_sub.param_names == ["a", "b", "f"]
+
+        # 4) Plain compiled(...) refuses — user must select an orch explicitly
+        a = torch.zeros((128, 128), dtype=torch.float32)
+        with pytest.raises(TypeError, match="select one explicitly"):
+            compiled(a, a, a)
+
+        # 5) Unknown orch name raises KeyError listing what *is* available
+        with pytest.raises(KeyError, match="chip_orch_add.*chip_orch_sub"):
```

Ruff (0.15.13)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)